### ISSUE TITLE
Equivalent of `let` in lisp?

### ISSUE THREAD

**Issue by kindlychung** [zatonovo/lambda.r#24]
Here is an example:

    ColorRGBhex(hexcode) %::% character : .
    ColorRGBhex(hexcode) %when% {
        length(hexcode) == 1
    } %as% {
        ColorRGBhex_(tolower(hexcode))
    }
    ColorRGBhex_(hexcode) %::% character : .
    ColorRGBhex_(hexcode) %when% {
        all(strsplit(hexcode, "")[[1]] >= "0")
        all(strsplit(hexcode, "")[[1]] <= "f")
    } %as% {
        Color_(hexcode)
    }

It would be nice if I can write it like this:

    ColorRGBhex(hexcode) %::% character : .
    ColorRGBhex(hexcode) %let% {
        hexcode = tolower(hexcode)
        hex_chars = strsplit(hexcode, "")[[1]]
    } %when% {
        length(hexcode) == 1
        all(hex_chars >= "0")
        all(hex_chars <= "f")
    } %as% {
        ColorRGBhex_(tolower(hexcode))
    }

This saves both typing and computation time.

**Comment by muxspace** [comment]
Great idea. Are you interested in contributing the implementation?

**Comment by kindlychung** [comment]
Yes, I am interested, but at the moment I don't really know lambda.r existing code well enough, it might take a while to read it thoroughly. I will stay tuned, it's a very nice package.

**Comment by muxspace** [comment]
There's really only one entrypoint, which is UseFunction. You can step through with the debugger and see how various constraints get built up.

**Comment by vixr** [comment]
any advantage over using built-in with()?
